File: japiwznm/WznmQRelList.java
/**
  * \file WznmQRelList.h
  * Java API code for record of table TblWznmQRelList
  * \author Alexander Wirthmueller
  * \date created: 25 Aug 2020
  * \date modified: 25 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQRelList {
	
	public WznmQRelList(
				int jnum
				, String stubFrRefWznmMTable
				, String stubFrsRefWznmMSubset
				, String stubToRefWznmMTable
				, String stubTosRefWznmMSubset
				, String srefIxVBasetype
				, String titIxVBasetype
				, String stubRefWznmMVersion
				, String stubSupRefWznmMRelation
				, String srefSupIxVSubrole
				, String titSupIxVSubrole
				, String stubRefWznmMTable
				, String Prefix
			) {
		this.jnum = jnum;
		this.stubFrRefWznmMTable = stubFrRefWznmMTable;
		this.stubFrsRefWznmMSubset = stubFrsRefWznmMSubset;
		this.stubToRefWznmMTable = stubToRefWznmMTable;
		this.stubTosRefWznmMSubset = stubTosRefWznmMSubset;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.stubRefWznmMVersion = stubRefWznmMVersion;
		this.stubSupRefWznmMRelation = stubSupRefWznmMRelation;
		this.srefSupIxVSubrole = srefSupIxVSubrole;
		this.titSupIxVSubrole = titSupIxVSubrole;
		this.stubRefWznmMTable = stubRefWznmMTable;
		this.Prefix = Prefix;
	};

	public int jnum;
	public String stubFrRefWznmMTable;
	public String stubFrsRefWznmMSubset;
	public String stubToRefWznmMTable;
	public String stubTosRefWznmMSubset;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String stubRefWznmMVersion;
	public String stubSupRefWznmMRelation;
	public String srefSupIxVSubrole;
	public String titSupIxVSubrole;
	public String stubRefWznmMTable;
	public String Prefix;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQRelList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubFrRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubFrRefWznmMTable", "frt", null, 0);
			stubFrsRefWznmMSubset = Xmlio.extractStringUclc(doc, basexpath, "stubFrsRefWznmMSubset", "frs", null, 0);
			stubToRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubToRefWznmMTable", "tot", null, 0);
			stubTosRefWznmMSubset = Xmlio.extractStringUclc(doc, basexpath, "stubTosRefWznmMSubset", "tos", null, 0);
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			stubRefWznmMVersion = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVersion", "ver", null, 0);
			stubSupRefWznmMRelation = Xmlio.extractStringUclc(doc, basexpath, "stubSupRefWznmMRelation", "sup", null, 0);
			srefSupIxVSubrole = Xmlio.extractStringUclc(doc, basexpath, "srefSupIxVSubrole", "srl", null, 0);
			titSupIxVSubrole = Xmlio.extractStringUclc(doc, basexpath, "titSupIxVSubrole", "srl2", null, 0);
			stubRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMTable", "tbl", null, 0);
			Prefix = Xmlio.extractStringUclc(doc, basexpath, "Prefix", "pfx", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQRelList.cpp
/**
	* \file WznmQRelList.cpp
	* API code for table TblWznmQRelList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 25 Aug 2020
	* \date modified: 25 Aug 2020
	*/

#include "WznmQRelList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQRelList
 ******************************************************************************/

WznmQRelList::WznmQRelList(
			const uint jnum
			, const string stubFrRefWznmMTable
			, const string stubFrsRefWznmMSubset
			, const string stubToRefWznmMTable
			, const string stubTosRefWznmMSubset
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string stubRefWznmMVersion
			, const string stubSupRefWznmMRelation
			, const string srefSupIxVSubrole
			, const string titSupIxVSubrole
			, const string stubRefWznmMTable
			, const string Prefix
		) {
	this->jnum = jnum;
	this->stubFrRefWznmMTable = stubFrRefWznmMTable;
	this->stubFrsRefWznmMSubset = stubFrsRefWznmMSubset;
	this->stubToRefWznmMTable = stubToRefWznmMTable;
	this->stubTosRefWznmMSubset = stubTosRefWznmMSubset;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->stubRefWznmMVersion = stubRefWznmMVersion;
	this->stubSupRefWznmMRelation = stubSupRefWznmMRelation;
	this->srefSupIxVSubrole = srefSupIxVSubrole;
	this->titSupIxVSubrole = titSupIxVSubrole;
	this->stubRefWznmMTable = stubRefWznmMTable;
	this->Prefix = Prefix;
};

bool WznmQRelList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQRelList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubFrRefWznmMTable", "frt", stubFrRefWznmMTable);
		extractStringUclc(docctx, basexpath, "stubFrsRefWznmMSubset", "frs", stubFrsRefWznmMSubset);
		extractStringUclc(docctx, basexpath, "stubToRefWznmMTable", "tot", stubToRefWznmMTable);
		extractStringUclc(docctx, basexpath, "stubTosRefWznmMSubset", "tos", stubTosRefWznmMSubset);
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVersion", "ver", stubRefWznmMVersion);
		extractStringUclc(docctx, basexpath, "stubSupRefWznmMRelation", "sup", stubSupRefWznmMRelation);
		extractStringUclc(docctx, basexpath, "srefSupIxVSubrole", "srl", srefSupIxVSubrole);
		extractStringUclc(docctx, basexpath, "titSupIxVSubrole", "srl2", titSupIxVSubrole);
		extractStringUclc(docctx, basexpath, "stubRefWznmMTable", "tbl", stubRefWznmMTable);
		extractStringUclc(docctx, basexpath, "Prefix", "pfx", Prefix);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQRelList
 ******************************************************************************/

ListWznmQRelList::ListWznmQRelList() {
};

ListWznmQRelList::ListWznmQRelList(
			const ListWznmQRelList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQRelList(*(src.nodes[i]));
};

ListWznmQRelList& ListWznmQRelList::operator=(
			const ListWznmQRelList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQRelList(*(src.nodes[i]));

	return *this;
};

ListWznmQRelList::~ListWznmQRelList() {
	clear();
};

void ListWznmQRelList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQRelList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQRelList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQRelList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQRelList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQRelList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQRelList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

